### DISCUSSION TITLE
How to place files other than executable files in a subdirectory while keeping the current path unchanged?

### DISCUSSION THREAD

**Discussion by JerryJian** [nulastudio/NetCoreBeauty#60] [Q&A]
Hi, I want to place all files without app.exe into a subfolder, but keep the `AppDomain.CurrentDomain.BaseDirectory` value as the exe folder.

I have tried following configurations, but the `AppDomain.CurrentDomain.BaseDirectory` has changed.

``` xml
	<PropertyGroup>
		<BeautyLibsDir>.</BeautyLibsDir>
		<BeautyOnPublishOnly>False</BeautyOnPublishOnly>
		<BeautyUsePatch>True</BeautyUsePatch>
		<BeautyAppHostEntry>app/Entry.dll</BeautyAppHostEntry>
		<BeautyAppHostDir>..</BeautyAppHostDir>
		<BeautyLogLevel>Info</BeautyLogLevel>
	</PropertyGroup>
```

**Comment by liesauer** [comment]
no, you can't. after moving the entry dll, the only stable way to get the correct exe folder is

`Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)`

**Reply by JerryJian** [reply]
Thanks

**Comment by JerryJian** [comment]
Sorry, that the configuration might meet my requirement, I check the following values:
```
Environment.ProcessPath: Publish\win\Entry.exe
AppContext.BaseDirectory: Publish\win\app
AppDomain.CurrentDomain.BaseDirectory: Publish\win\app
Environment.CurrentDirectory: Publish\win
Directory.GetCurrentDirectory(): Publish\win
```
I will close this discussion.
